src/passes/TranslateEH.cpp

```diff
@@ -0,0 +1,823 @@
+    //     ;; already translated to try_table at this point. See
+    //     ;; processDelegate() for how it is done.
+    //     (try_table (catch_all_ref $delegate_br_target)
+    //       ...
+    //     )
+    //     ...
+    //   (catch
+    //     ...
+    //   )
+    // )
+    //
+    // to =>
+    //
+    // If sometype (try's type) is none:
+    // (block $outer (result sometype)
+    //   (try (result sometype)
+    //     (do
+    //       (throw_ref
+    //         (block $delegate_br_target (result exnref)
+    //           ...
+    //           (try_table (catch_all_ref $delegate_br_target)
+    //             ...
+    //           )
+    //           ...
+    //           (br $outer)
+    //         )
+    //       )
+    //     )
+    //     (catch
+    //       ...
+    //     )
+    //   )
+    // )
+    //
+    // If sometype (try's type) is concrete:
+    // (block $outer (result sometype)
+    //   (try (result sometype)
+    //     (do
+    //       (throw_ref
+    //         (block $delegate_br_target (result exnref)
+    //           (br $outer
+    //             ...
+    //             (try_table (catch_all_ref $delegate_br_target)
+    //               ...
+    //             )
+    //             ...
+    //           )
+    //         )
+    //       )
+    //     )
+    //     (catch
+    //       ...
+    //     )
+    //   )
+    // )
+    //
+    // Note that the current try-catch (or try-delegate) stays as is for now; it
+    // will be converted to a try_table later in processDelegate() and
+    // processCatches().
+    //
+    // Also note that even in case there are multiple inner try~delegates
+    // targetting this try, we need to do this only once per try target. Those
+    // multiple try~delegates that used to target the same delegate target now
+    // jump to the same $delegate_br_target using catch_all_ref.
+    Name delegateBrTarget = delegateTargetToBrTarget[curr->name];
+    Expression* innerBody = nullptr;
+    if (curr->type.isConcrete()) {
+      auto* brToOuter = builder.makeBreak(outerBlock->name, curr->body);
+      innerBody = builder.blockifyWithName(
+        brToOuter, delegateBrTarget, nullptr, Type(HeapType::exn, Nullable));
+    } else {
+      auto* brToOuter = curr->body->type == Type::unreachable
+                          ? nullptr
+                          : builder.makeBreak(outerBlock->name);
+      innerBody = builder.blockifyWithName(
+        curr->body, delegateBrTarget, brToOuter, Type(HeapType::exn, Nullable));
+    }
+    curr->body = builder.makeThrowRef(innerBody);
+    assert(outerBlock->list.empty());
+    outerBlock->list.push_back(curr->body);
+  }
+
+  void processDelegate(Try* curr) {
+    Builder builder(*getModule());
+    // Convert
+    // (try
+    //   (do
+    //     ...
+    //   )
+    //   (delegate $delegate_target)
+    // )
+    //
+    // to =>
+    //
+    // (try_table (catch_ref $delegate_br_target)
```

Comment:
Technically yes in the actual code, because we are getting a unique label every time anyway so that if labels overlap they be affixed by a number. But I'm worried combining them in this example in the comment makes things confusing, because in the 'before' example, we actually have two different labels at that point, one being the old outer `try` (which has yet to be converted to `try_table`'s delegate target label, and the other being the inner `try_table`'s branch target:
```cpp
    // (try $delegate_target (result sometype)
    //   (do
    //     ...
    //     ;; This had originally been an inner try~delegate and has been
    //     ;; already translated to try_table at this point. See
    //     ;; processDelegate() for how it is done.
    //     (try_table (catch_all_ref $delegate_br_target)
    //       ...
    //     )
    //     ...
    //   (catch
    //     ...
    //   )
```

These two are different labels and we have a map to convert one to the other (`delegateTargetToBrTarget`), so I'n worried if using the same labels in the example (and thus in the code) will make things confusing.